File: 792-binary-search/binary-search.py
import math

class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums) - 1

        while left <= right:
            mid = math.floor((right + left) / 2)
            # print(f"Left: {left} Right: {right} Mid: {mid}")
            # print(f"Left Item: {nums[left]} Right Item: {nums[right]} Mid Item: {nums[mid]}")
            if nums[mid] == target:
                return mid
            elif target < nums[mid]:
                right = mid - 1
            else:
                left = mid + 1
        return -1

File: 792-binary-search/binary-search.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {

      int   l = 0;
      int  r = nums.size() - 1;

        while(l <= r){
            if (target < nums[l] || target > nums[r])  return -1;
            if (target == nums[r])  return r;
            int m = l + floor(((target - nums[l])/(nums[r] - nums[l])) * (r - l));
            if (nums[m] == target){
                return m;
             } else if (nums[m] < target) {
                l = m + 1;
              } else {
                r = m - 1;
              }
        }
        return -1;
    }
};